test/noflo-component-factory-mocha.js

```diff
@@ -32,11 +32,114 @@ describe('noflo-component-factory', function() {
             component.inPorts.input.detach(socket);
         }).should.become("hello");
     });
+    it("should behave like an EventEmitter", function() {
+        return new Promise(function(done, fail){
+            var handler = function(payload){
+                done(payload + ' ' + this.name);
+            };
+            var component = test.createComponent(componentFactory({
+                inPorts:{
+                    input1:{
+                        ondata: function(payload) {
+                            this.on('change', fail);
+                            this.removeListener('change', fail);
+                            this.on('change', handler);
+                        }
+                    },
+                    input2:{
+                        ondata: function(payload) {
+                            this.on('change', handler);
+                            this.emit('change', payload);
+                        }
+                    }
+                }
+            }));
+            var input1 = noflo.internalSocket.createSocket();
+            component.inPorts.input1.attach(input1);
```

Comment:
Can you use the common-test.sendData function so we don't keep duplicating all this code?  It's less to write and read.